example/provider-extensions/gardener-discovery-server/deploy.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+#
+# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+REPO_ROOT_DIR="$(realpath "$SCRIPT_DIR"/../../..)"
+
+usage() {
+  echo "Usage:"
+  echo "> ${0} [ -h | <garden_kubeconfig> <seed_kubeconfig> ]"
+  echo
+  echo ">> For example: ${0} ~/.kube/garden-kubeconfig.yaml ~/.kube/seed-kubeconfig.yaml"
+
+  exit 0
+}
+
+if [ "$1" == "-h" ] || [ "$#" -ne 2 ]; then
+  usage
+fi
+
+garden_kubeconfig=$1
+seed_kubeconfig=$2
+
+kubectl --server-side=true --kubeconfig "$garden_kubeconfig" apply -f "$SCRIPT_DIR"/rbac
+
+token=$(kubectl --kubeconfig "$garden_kubeconfig" -n garden create token gardener-discovery-server --duration 24h)
```

Comment:
To be fair I think that the discovery server should continue to work even without that kubeconfig. It serves the workload identity documents that it reads from a secret that is mounted to the filesystem of the container. The thing that will stop working is the controller that reads the discovery documents of shoot clusters which is something that is not supported in this dev scenario because the configuration is read from a secret labeled with `gardener.cloud/role: shoot-service-account-issuer` and that one is not configured properly in this setup (see [an example](https://github.com/gardener/gardener/blob/master/example/gardener-local/controlplane/service-account-issuer-secret.yaml)).

> We should at least mention that in getting_started_locally_with_extensions.yaml or create a script which updates the kubeconfig and restarts gardener-discovery-server, shouldn't we?

Yes, we can mention it in the docu file. Do you think that it is also beneficial to increase the token validity to 48h?

Reply:
Let me confirm that on Monday and will unhold the PR if that is really true 😄